modules/core/04-channel/keeper/keeper.go

```diff
@@ -824,3 +826,29 @@ func (k *Keeper) PruneAcknowledgements(ctx context.Context, portID, channelID st
 
 	return totalPruned, totalRemaining, nil
 }
+
+// GetV2Counterparty returns a version 2 counterparty for the given port and channel ID
+// by converting the channel into a version 2 counterparty
+func (k *Keeper) GetV2Counterparty(ctx context.Context, portID, channelID string) (packetserver.Counterparty, bool) {
+	channel, ok := k.GetChannel(ctx, portID, channelID)
+	if !ok {
+		return packetserver.Counterparty{}, false
+	}
+	// Do not allow channel to be converted into a version 2 counterparty
+	// if the channel is not OPEN or if it is ORDERED
+	if channel.State != types.OPEN || channel.Ordering == types.ORDERED {
+		return packetserver.Counterparty{}, false
+	}
+	connection, ok := k.connectionKeeper.GetConnection(ctx, channel.ConnectionHops[0])
+	if !ok {
+		return packetserver.Counterparty{}, false
+	}
+	merklePathPrefix := commitmentv2types.NewMerklePath(connection.Counterparty.Prefix.KeyPrefix, []byte(""))
+
+	counterparty := packetserver.Counterparty{
+		CounterpartyChannelId: channelID,
```

Comment:
```suggestion
		CounterpartyChannelId: channel.Counterparty.ChannelId,
```

This needs to be storing the counterparty's channelID not our own.